plugins/hydra_nevergrad_sweeper/tests/test_nevergrad_sweeper_plugin.py

```diff
@@ -43,8 +43,10 @@ def test_discovery() -> None:
         ),
     ],
 )
-def test_make_parameter(string: str, param_cls: Any, value_cls: Any) -> None:
-    param = make_parameter(string)
+def test_make_parameter_from_commandline(
+    string: str, param_cls: Any, value_cls: Any
+) -> None:
+    param = core.make_parameter_from_commandline(string)
```

Comment:
Do you allow the config to define things not defined in the command line?
In the Ax plugin, the command line is not flexible enough for all of the options, but some of the options like logscale are per variable and does not change often.
to support them (and also to force the type to int or float etc), the user can add that variable to the params block in the plugin config and describe it (logscale, bounds, type etc).
The command line can change the bounds, and can influence the inferred type but the final parameter description is a combination if the config, the command line and the inferred type (if the config is not specifying it - I believe (@shagunsodhani, is my description accurate?)).

Reply:
The config can define things not defined in the command line indeed, however the commandline completely overrides the config. That was simpler to implement and seems a bit more natural to me

Reply:
That makes sense in general, however inferred things are different:
they are not explicitly defined in the command line. so if they are explicitly defined in the config the config should win.

in other words, if I config says the type is a float and the user pass `foo=0:1` I think it should be interpreted as a float.
if the config says the type is an int and the user passes `foo=0.0:1.0`, this should be an error as clearly the user is misunderstanding something.

| config | Inferred from CLI  | Interpretation |
| ----------|-----|------|
| -               | int    | int   |
| -               | float | float|
| int             | int    | int   |
| int             | float | error |
| float          | int    | float |
| float          | float | float |

We can come up with a syntax to be explicit about the type in the command line, maybe
`foo=float:0:1`, which can also support logscale toggling. if we had that, that should win over the config (I don't think we need this just yet though).

Thoughts (cc @shagunsodhani)

Reply:
Information specified explicitly in the command line should _always_ win.
Inferred things are not specified explicitly and as such I view them as lower priority to the config (which is another thing the user specified explicitly).

If we default to float we can no longer support inferring the type.
It means the only option is to use `int:0:1` to enforce an int, or to specify int in the config for this variable.

I am okay with it though. I think this whole think of inferring the type from the command line is a bit shady.

Reply:
> I am okay with it though. I think this whole think of inferring the type from the command line is a bit shady.

Yes, I dont like it at all :s

Reply:
Who wants to write down a proposal for how we should do all this generally?
we can implement this proposal for both plugins and make it the recommendation for other HPO sweeper plugins.

Reply:
We can edit it together, just want to get some initial draft to work on.

Reply:
in my opinion the proposition above would fit all cases i need
```
# bounds (min:max)
0:1           # specs are optional, assumed float if int not provided
int:0:1
log:0.001:1 (log 0 somehow means unbounded...)
log:int:0.001:1
int:log:0.001:1 # order of specs does not matter

# choices
anything,whatever,something
```

Reply:
or do you mean even more general specs like:
`init:3:step:2:lower:0:upper10`?

Reply:
I am talking about the whole thing.
How we configure, should the command like look like.
How are the two interacting.
What should trigger errors (like the the table I suggested above).

The proposition above can be a part of that proposal.

Once we agree on what to do we can even share some of the implementations.
for example, we can have a function that takes the params node, the arguments the sweeper is taking, and convert it to a generic intermediate representation describing each variable.
That representation can be converted to plugin specific representation by each of the plugins.
The advantage is that this will enable to consistent behavior between the two plugins and will reduce duplicated effort.

Reply:
agreed. Given I've never been a fan of commandlines, I may not be the best one to answer this :D

Just a couple of thoughts to add:
- ranges seem the easiest way to work in commandline, but it does not cover everything, like the one example above where I want to provide an initial value, and an a gaussian prior (standard deviation / step), and where I could want to remove the upper and lower bounds.
- related to this is the fact that what a framework can handle, the other may not, so having common spec may be difficult
 
For now I was thinking of only providing ranges in commandline, and users would have more options if they use the config file. The generic intermediate specs would be nice but may require more time to think about that I can currently have.

Reply:
for ax, command line can now specify ranges (cont or discrete), choices, and fixed (constant values).
anything else (logscale, explicit types specification) requires the config file.
if a framework does not support something in the common spec it can throw an exception if it gets a request to do that.  think it's 
to override very specialized things from the command line you can always override the config node directly:
`hydra.sweeper.params.spec.z.std=2.0`

(spec seems like a decent name for the specifications. at least it's not params.params :) ).

unrelated idea I had: we can think outside of the box and specify the modifiers before the = :
`x=0.0:1.0` # deduced float
`x:float=0:1` # explicit float
`x:log,float=0:1` # explicit float and logscale.

Reply:
`x:log,float=0:1`
I agree this could be convenient, but may be a bit confusing

I'm fine with `spec` as well, I don't have much preference

Reply:
Let's discuss both the points (spec and command line syntax over the other issue talking about unifying the interface of the two plugins.